From def5f0d7143dd0d813700b8b0a1b019fe91ba5d4 Mon Sep 17 00:00:00 2001
From: Wei Sun <wei.w.sun@intel.com>
Date: Mon, 29 Jun 2015 11:46:32 +0800
Subject: [PATCH 435/462] atomisp2: use atomisp_update_capture_mode to select
 capture mode

Use atomisp_update_capture_mode which will refer to previous setting
to select capture mode.

Change-Id: I26e0aa802b119f226b4c5c687140053db043cf89
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-38327
Signed-off-by: Wei Sun <wei.w.sun@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/62568
Approver: Bin B Han <bin.b.han@intel.com>
Reviewed-by: Bin B Han <bin.b.han@intel.com>
Build: IMIN (SC)
Integrator: WeiX Meng <weix.meng@intel.com>
Validation-Android: Wei Tao <wei.tao@intel.com>
Build: WeiX Meng <weix.meng@intel.com>
Reviewed-by: WeiX Meng <weix.meng@intel.com>
Maintainer: WeiX Meng <weix.meng@intel.com>
---
 .../camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c      | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
index 05f0548..5cb20de 100755
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
@@ -4697,7 +4697,7 @@ static int __enable_continuous_mode(struct atomisp_sub_device *asd,
 		"continuous mode %d, raw buffers %d, stop preview %d\n",
 		enable, asd->continuous_raw_buffer_size->val,
 		!asd->continuous_viewfinder->val);
-	atomisp_css_capture_set_mode(asd, CSS_CAPTURE_MODE_PRIMARY);
+	atomisp_update_capture_mode(asd);
 	/* in case of ANR, force capture pipe to offline mode */
 	atomisp_css_capture_enable_online(asd, ATOMISP_INPUT_STREAM_GENERAL,
 			asd->params.low_light ? false : !enable);
-- 
1.9.1

